test/gtest/ucs/test_lru.cc

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2001-2023. ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+
+extern "C" {
+#include "ucs/datastruct/lru.h"
+}
+
+class test_lru : public ucs::test {
+protected:
+    test_lru() : m_lru(NULL)
+    {
+    }
+
+    virtual void init()
+    {
+        ucs::test::init();
+        ASSERT_UCS_OK(ucs_lru_create(m_capacity, &m_lru));
+    }
+
+    virtual void cleanup()
+    {
+        ucs_lru_destroy(m_lru);
+        ucs::test::cleanup();
+    }
+
+    void run(std::vector<int> &elements)
+    {
+        for (size_t i = 0; i < m_capacity * 10; ++i) {
+            ucs_lru_put(m_lru, &elements[i % elements.size()]);

```

Comment:
The different tests play with capacity, but basically the scenario is the same - precise round-robin across all elements.
Consider adding something else which would challenge the essence of the LRU.

Reply:
Add LRU size check, as we discussed.